go/tasks/plugins/k8s/dask/dask.go

```diff
@@ -0,0 +1,300 @@
+
+	return daskAPI.DaskJobSpec{
+		Job: daskAPI.JobSpec{
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					jobContainer,
+				},
+			},
+		},
+		Cluster: daskAPI.JobClusterSpec{
+			Spec: daskAPI.DaskClusterSpec{
+				Worker:    workerSpec,
+				Scheduler: schedulerSpec,
+			},
+		},
+	}
+}
+
+func (p daskResourceHandler) GetTaskPhase(ctx context.Context, pluginContext k8s.PluginContext, r client.Object) (pluginsCore.PhaseInfo, error) {
+	logPlugin, err := logs.InitializeLogPlugins(logs.GetLogConfig())
+	if err != nil {
+		return pluginsCore.PhaseInfoUndefined, err
+	}
+
+	return p.GetTaskPhaseWithLogs(ctx, pluginContext, r, logPlugin, " (User)")
+}
+
+func (daskResourceHandler) GetTaskPhaseWithLogs(ctx context.Context, pluginContext k8s.PluginContext, r client.Object, logPlugin tasklog.Plugin, logSuffix string) (pluginsCore.PhaseInfo, error) {
+	job := r.(*daskAPI.DaskJob)
+
+	pod, err := getJobPodFromJobResource(job, ctx)
+	if err != nil {
+		return pluginsCore.PhaseInfoUndefined, err
+	}
+	
+	// Logic copied from the pod plugin
+	transitionOccurredAt := flytek8s.GetLastTransitionOccurredAt(pod).Time
+	info := pluginsCore.TaskInfo{
+		OccurredAt: &transitionOccurredAt,
+	}
+
+	if pod.Status.Phase != v1.PodPending && pod.Status.Phase != v1.PodUnknown {
+		taskLogs, err := logs.GetLogsForContainerInPod(ctx, logPlugin, pod, 0, logSuffix)
+		if err != nil {
+			return pluginsCore.PhaseInfoUndefined, err
+		}
+		info.Logs = taskLogs
+	}
+
+	switch pod.Status.Phase {
+	case v1.PodSucceeded:
+		return flytek8s.DemystifySuccess(pod.Status, info)
+	case v1.PodFailed:
+		return flytek8s.DemystifyFailure(pod.Status, info)
+	case v1.PodPending:
+		return flytek8s.DemystifyPending(pod.Status)
+	case v1.PodReasonUnschedulable:
+		return pluginsCore.PhaseInfoQueued(transitionOccurredAt, pluginsCore.DefaultPhaseVersion, "pod unschedulable"), nil
+	case v1.PodUnknown:
+		return pluginsCore.PhaseInfoUndefined, nil
+	}
+
+	if len(info.Logs) > 0 {
+		return pluginsCore.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion+1, &info), nil
+	}
+	return pluginsCore.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion, &info), nil
+}
+
+
+func getClientset() (*kubernetes.Clientset, error) {
+	// FIXME: Handle auth
+	config, err := clientcmd.BuildConfigFromFlags("", "/Users/bstadlbauer/.kube/config")
+	if err != nil {
+		return nil, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return clientset, nil
+}
+
+func getJobPodFromJobResource(job *daskAPI.DaskJob, ctx context.Context) (*v1.Pod, error) {
+	clientset, err := getClientset()
+	if err != nil {
+		return nil, err
+	}
+
+	jobPodName := job.ObjectMeta.Name + jobRunnerPodPostfix
+	jobPodNamespace := job.ObjectMeta.Namespace
+	pod, err := clientset.CoreV1().Pods(jobPodNamespace).Get(ctx, jobPodName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return pod, nil
+}
```

Comment:
At the moment, the `DaskJob` CR has no notion of a state. So to get the state, I am currently instantiating a k8s `clientset`, query for the job pod, and use that to determine state, similar to how the `pod` plugin does that.\

My questions here would be: 
1. Is this something that is acceptable? From what I've seen, usually only propeller has a notion of `clientset`, other plugins rely on the state within the CR provided by to the plugin by the propeller. If this is not an option, I can try to contribute to the `dask`-operator, potentially we could save some state in there.
2. If yes, should there be configuration which allows authenticating using a local kubeconfig, or should we only support in-cluster mode?

Reply:
ideally plugins should not create a new k8s config. This context is better kept within propeller and injected in.

Is it that you want to get the reference to the driver pod? We can do that in a different way